postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -220,11 +274,18 @@ def _collect_statement_samples(self):
         start_time = time.time()
         rows = self._get_new_pg_stat_activity()
         rows = self._filter_valid_statement_rows(rows)
-        events = self._explain_pg_stat_activity(rows)
+        event_samples, activity_event = self._collect_events(rows)
```

Comment:
We may want to structure this a bit differently. 

```python
rows = self._normalize_rows(rows)
for plan in self._collect_plans(rows):
   submit plan
if it's time to collect activity:
   active_connections = self._get_active_connections()
   event = self._create_activity_event(rows, active_connections)
   submit event
```

Reasons:
* `normalize_rows` would calculate the query signature and obfuscate the query. @alexandre-normand added this pattern for statement metrics and I think it's a good idea to do the same here. Then the normalized rows can be reused for both plans and activity separately. 
* plan collection discards events that fail obfuscation, but for activity we don't want to discard those because if a query fails obfuscation but it's doing something important, we can still ingest the rest of the activity data, and have the query field be "ERROR: failed to obfuscate" or something. If the one query that has a stuck transaction is the one that failed to obfuscate, we don't want to lose that. 
* decouple the plan collection code from the activity code

Reply:
> plan collection discards events that fail obfuscation, but for activity we don't want to discard those because if a query fails obfuscation we can still ingest the rest of the activity data, and have the query field be something like ERROR: failed to obfuscate. If the one query that has a stuck transaction is the one that failed to obfuscate, we don't want to lose that activity info.

in order to send the query to the backend, isn't it a requirement that we obfuscate it?

Reply:
oh I am dumb today you already answered this
 
>  if a query fails obfuscation we can still ingest the rest of the activity data, and have the query field be something like ERROR: failed to obfuscate

Reply:
https://github.com/DataDog/integrations-core/pull/9866/commits/fdbf42c7e50cdc03332776e58bced9f1cd08c9fd